### PR TITLE
publish to npmjs workflow

### PR DESCRIPTION
**Pull request by ThomasGoatITMedia**
In this PR I've updated the readme with documentation about how to release a new version and I've created a GitHub workflow that should automagically release on npmjs as well.

In previous release the version number wasn't pushed due to the protected main/master branch. This PR fixes the version bump and when following the steps should fix that in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,31 @@ rules: {
...
+## Publishing to npmjs.com
+
+### prerequisites
+
+You’ll need an account on npmjs.com and that has access to this package.
+Account needs 2FA enabled for authorization & publishing.
+
+### Step 1 - Bump version
+
+Create a new branch in which you bump the version, in here you'll also need to update [CHANGELOG.md](https://github.com/Mobiliteitsfabriek/eslint-config-mig-react-native/blob/main/CHANGELOG.md). Move everything from [Unreleased] to the new version that you're creating. Using `npm version` you can bump the version.
```
Comment (aymkin) [inline README.md]: it would be great to have the naming pattern for branches, for example, `version-bump/*`

**Review by aymkin — DISMISSED** [review]
Just a tiny comment

File: README.md (hunk 1 above)
Comment (ThomasGoatITMedia) [reply README.md]: Good idea! I will add it to the documentation

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
+          registry-url: https://npm.pkg.github.com/
+      - run: npm ci
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (isilher) [inline .github/workflows/npm-publish.yml]: I don't think we need to publish to GitHub Packages necessarily.

File: .github/workflows/npm-publish.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
+      - run: npm ci
+      - run: npm test
```
Comment (isilher) [inline .github/workflows/npm-publish.yml]: Why create a separate job (with checkout and install) just to run the tests? We could add just the test step to the publish job.

**Review by isilher — CHANGES_REQUESTED** [review]
Very nice! I have two comments but I in general this is a great setup.

**Comment by isilher** [comment]
@ThomasGoatITMedia removing the CI step will make the test job fail. I was proposing to merge the two jobs, removing the duplicate steps. I have made a commit to do that, feel free to revert if you disagree :-)

**Comment by ThomasGoatITMedia** [comment]
@isilher wasn't completely sure what you meant haha :) it's good now

**Review by isilher — APPROVED** [review]
👍

**Review by aymkin — APPROVED** [review]

**Pull request merged**